src/lib/strings/url-helpers.ts

```diff
@@ -170,15 +170,36 @@ export function getYoutubeVideoId(link: string): string | undefined {
 
 export function linkRequiresWarning(uri: string, label: string) {
   const labelDomain = labelToDomain(label)
-  if (!labelDomain) {
-    return true
-  }
+  let urip
   try {
-    const urip = new URL(uri)
-    return labelDomain !== urip.hostname
+    urip = new URL(uri)
   } catch {
     return true
   }
+
+  if (urip.hostname === 'bsky.app') {
+    // if this is a link to internal content,
+    // warn if it represents itself as a URL to another app
```

Comment:
what if it's a link to bsky but a different page than claimed?

Reply:
this is the meat of it here. if you look at the issue that triggered this, people do thinks like "see [this thread]". the question is what exactly do we want to mitigate, and how

a likely issue would be faking mentions, like doing "@bob.xyz" and it actually goes to "@bob.site". right now that's not getting covered. after that it's just misleading labels like "this post by paul frazee" and it's a post by "@pfrazee.io." I'd rather not do an interstitial on *every* link to internal content that's not a raw URL, so I'm not sure what kinds of mitigations to aim for there